app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt

```diff
@@ -0,0 +1,338 @@
+ * @param description An optional description text below the label.
+ * @param onClick Called when the user clicks on the item.
+ * @param drawable [DrawableRes] used to display a drawable at the start of the list item.
+ * @param drawableDescription Content description of the drawable.
+ * @param icon Optional [DrawableRes] used to display a drawable at the end of the list item.
+ * @param iconDescription Content description of the icon.
+ * @param onIconClick Called when the user clicks on the icon.
+ */
+@Composable
+fun IconListItem(
+    label: String,
+    description: String? = null,
+    onClick: (() -> Unit)? = null,
+    @DrawableRes drawable: Int,
+    drawableDescription: String? = null,
+    @DrawableRes icon: Int? = null,
+    iconDescription: String? = null,
+    onIconClick: (() -> Unit)? = null,
+) {
+    ListItem(
+        label = label,
+        description = description,
+        onClick = onClick,
+        beforeListAction = {
+            ListItemIcon(
+                icon = drawable,
+                iconDescription = drawableDescription,
+                onClick = null,
+            )
+        },
+        afterListAction = {
+            icon?.let {
+                ListItemIcon(
+                    icon = icon,
+                    iconDescription = iconDescription,
+                    onClick = onIconClick,
+                )
+            }
+        }
+    )
+}
+
+/**
+ * Base list item used to display a label with an optional description text and
+ * the flexibility to add custom UI to either end of the item.
+ *
+ * @param label The label in the list item.
+ * @param description An optional description text below the label.
+ * @param onClick Called when the user clicks on the item.
+ * @param beforeListAction Optional Composable for adding UI to the start of the list item.
+ * @param afterListAction Optional Composable for adding UI to the end of the list item.
+ */
+@Composable
+private fun ListItem(
+    label: String,
+    description: String? = null,
+    onClick: (() -> Unit)? = null,
+    beforeListAction: @Composable RowScope.() -> Unit = {},
+    afterListAction: @Composable RowScope.() -> Unit = {},
+) {
+    Row(
+        modifier = when (onClick != null) {
+            true -> Modifier.clickable { onClick() }
+            false -> Modifier
+        }.then(
+            Modifier.defaultMinSize(minHeight = LIST_ITEM_HEIGHT)
+        ),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        beforeListAction()
+
+        Column(
+            modifier = Modifier
+                .padding(horizontal = 16.dp, vertical = 10.dp)
+                .weight(1f),
+        ) {
+            PrimaryText(
+                text = label,
+                fontSize = 16.sp,
+                maxLines = 1,
+            )
+
+            description?.let {
+                SecondaryText(
+                    text = description,
+                    fontSize = 12.sp,
```

Comment:
Just rechecked this based on padding comment since I was wondering if the label + description was actually getting 10.dp padding vertically since it looked a bit big compare to 6.dp. This should be 14.sp

Reply:
Thanks for catching this